src/main/java/com/google/step/snippet/servlets/HomepageServlet.java

```diff
@@ -25,9 +26,19 @@
 @WebServlet("/")
 public class HomepageServlet extends HttpServlet {
 
+  private static final String AUTH_URL = "authUrl";
+  private static final String AUTH_LABEL = "authLabel";
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
+    String redirectPath = request.getRequestURI();
+    if (request.getQueryString() != null) {
+      redirectPath += "?" + request.getQueryString();
+    }
```

Comment:
I am confused what this does. What does request.getQueryString() do? What is the intent of this code?

Reply:
request.getQueryString() pulls everything after the ? on the path. So in the case of /search?q=html it will pull q=html which I need to construct the full link to redirect to after. But I probably actually don't need this code specifically on the HomepageServlet which will always be '/'